lib/ansible/module_utils/common/respawn.py

```diff
@@ -66,6 +69,24 @@ def probe_interpreters_for_module(interpreter_paths, module_name):
         'PYTHONPATH': f'{_ANSIBLE_PARENT_PATH}:{PYTHONPATH}'.rstrip(': ')
     })
 
+    import_string = ''
+    if module_names is not None:
+        module_names.append('ansible.module_utils.basic')
+        import_string = f"import {', '.join(module_names)}"
+
+    if module_name is not None:
+        warnings.deprecate(
+            msg='Use of module_name is deprecated.',
+            version='2.25',
+            obj=module_name,
+            help_text='Use module_names instead to import multiple Python modules.',
+        )
+        module_import = f'import {module_name}, ansible.module_utils.basic'
+        if import_string:
+            import_string += f'; {module_import}'
+        else:
+            import_string = module_import
```

Comment:
How about creating a new list of import names, like this:
```suggestion
    import_modules = ['ansible.module_utils.basic']
    if module_names is not None:
        import_modules.extend(module_names)

    if module_name is not None:
        warnings.deprecate(
            msg='Use of module_name is deprecated.',
            version='2.25',
            obj=module_name,
            help_text='Use module_names instead to import multiple Python modules.',
        )
        import_modules.append(module_name)

    import_string = f"import {', '.join(import_modules)}"
```
This way it prevents adding `ansible.module_utils.basic` twice if both `module_names` and `module_name` is used, it does not modify `module_names`, and it allows to pass tuples or other kind of sequences as `module_names`.